src/hypothesis/core.py

```diff
@@ -586,16 +589,13 @@ def run_test_with_generator(test):
 
         arguments = original_argspec.args
 
-        for name, strategy in zip(
-            arguments[-len(generator_arguments):], generator_arguments
-        ):
+        for name, strategy in zip(arguments[-len(generator_arguments):],
+                                  generator_arguments):
             generator_kwargs[name] = strategy
 
-        argspec = inspect.ArgSpec(
+        argspec = original_argspec._replace(
```

Comment:
also it leaves the annotation for 'return', which is incorrect: A function decorated with given always returns None, regardless of what the wrapped function returns.